pymc3/distributions/bound.py

```diff
@@ -263,47 +75,61 @@ class Bound:
                     'par2', mu=0.0, sigma=1.0, initval=1.0)
     """
 
-    def __init__(self, distribution, lower=None, upper=None):
+    @classmethod
+    def dist(self, distribution, lower=None, upper=None):
         if isinstance(lower, Real) and lower == -np.inf:
             lower = None
         if isinstance(upper, Real) and upper == np.inf:
             upper = None
 
-        if not issubclass(distribution, Distribution):
-            raise ValueError('"distribution" must be a Distribution subclass.')
-
-        self.distribution = distribution
-        self.lower = lower
-        self.upper = upper
-
-    def __call__(self, name, *args, **kwargs):
-        if "observed" in kwargs:
-            raise ValueError(
-                "Observed Bound distributions are not supported. "
-                "If you want to model truncated data "
-                "you can use a pm.Potential in combination "
-                "with the cumulative probability function."
-            )
+        # if "observed" in kwargs:
+        #     raise ValueError(
+        #         "Observed Bound distributions are not supported. "
+        #         "If you want to model truncated data "
+        #         "you can use a pm.Potential in combination "
+        #         "with the cumulative probability function."
+        #     )
 
-        transform = kwargs.pop("transform", "infer")
-        if issubclass(self.distribution, Continuous):
-            return _ContinuousBounded(
-                name, self.distribution, self.lower, self.upper, transform, *args, **kwargs
-            )
-        elif issubclass(self.distribution, Discrete):
-            return _DiscreteBounded(
-                name, self.distribution, self.lower, self.upper, transform, *args, **kwargs
-            )
+        transform = kwargs.pop("transform", None)
+        if issubclass(distribution, Continuous):
+            if transform == None:
+                if lower is not None and upper is not None:
+                    transform = transforms.interval(lower, upper)
+                    default = 0.5 * (lower + upper)
+                elif upper is not None:
+                    transform = transforms.upperbound(upper)
+                else:
+                    transform = transforms.lowerbound(lower)
+
+        elif issubclass(distribution, Discrete):
+            raise ValueError("Can not transform discrete variable.")
```

Comment:
Not sure if `interval` is the right thing for `Discrete` distributions.

Reply:
What do you mean?

Reply:
In `v3` the implementation avoided use of `interval` transform upon `Discrete` (maybe because it uses `sigmoid` that would lead to values not in the allowed values of the discrete distribution) so I'm not sure about applying the same upon `Discrete`. What could be a good approach here ?

Reply:
I think you should follow the original, transform only continuous variables and not transform discrete ones. The bounding of discrete is achieved by adding a pm.Potential in the logp (which you should add for both discrete and continuous distributions anyway)